springfox-swagger2/src/main/java/springfox/documentation/swagger2/web/Swagger2Controller.java

```diff
@@ -63,29 +63,30 @@
   private ServiceModelToSwagger2Mapper mapper;
 
   @Autowired
-  private JsonSerializer jsonSerializer;
+  private MultiFormatSerializer multiFormatSerializer;
 
   @ApiIgnore
   @RequestMapping(value = "${springfox.documentation.swagger.v2.path:" + DEFAULT_URL + "}",
       method = RequestMethod.GET, produces = { APPLICATION_JSON_VALUE, HAL_MEDIA_TYPE })
   public
   @ResponseBody
-  ResponseEntity<Json> getDocumentation(
+  ResponseEntity<RawOutput> getDocumentation(
+      @RequestParam(value = "format", required = false) String format,
```

Comment:
The main reason is that when I usually "demo" the swagger docs i do it from the browser, not via a web client where I can set additional headers. 

The second reason why I did this was because yaml doesn't have any internet media type registered yet. This second reason is why i didn't change the produces header here as well, there is no standard for yaml as far as I'm aware of.